openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -432,15 +435,31 @@ def test_unsuccessful_get_account_by_user_id(self):
         """
         api_client = "client"
         user = "user"
+        url = reverse("accounts_detail_api")
         client = self.login_client(api_client, user)
         self.create_mock_profile(self.user)
         set_user_preference(self.user, ACCOUNT_VISIBILITY_PREF_KEY, PRIVATE_VISIBILITY)
 
-        response = self.send_get(
-            client, query_parameters=f'lms_user_id={self.user.id}', expected_status=status.HTTP_403_FORBIDDEN
-        )
+        response = client.get(url + f'?lms_user_id={self.user.id}')
+        assert response.status_code == status.HTTP_403_FORBIDDEN
         assert response.data.get('detail') == 'You do not have permission to perform this action.'
 
+    @ddt.data('abc', '2f', '1.0', "2/8")
```

Comment:
This test is ideally meant for `non-integer` values which include float, negative values and some other types. Please add non-positive integer, non-string values as well.

Reply:
obviously adding a non-string value eventually ends up the same e.g. both 1.0 and '1.0' will end up the same in the url.
Also, the null input is handled as well in the code (gets the requester's account info instead).